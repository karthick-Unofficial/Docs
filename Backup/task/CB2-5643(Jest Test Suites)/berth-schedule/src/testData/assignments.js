const assignments = {
    "allAssignments": {},
    "dailyAssignments": {}
};

export { assignments };