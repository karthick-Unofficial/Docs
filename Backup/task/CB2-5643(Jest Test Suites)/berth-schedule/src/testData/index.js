export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { map } from "./map";
export { berths } from "./berths";
export { formPanel } from "./formPanel";
export { assignments } from "./assignments";
export { clientConfig } from "./clientConfig";