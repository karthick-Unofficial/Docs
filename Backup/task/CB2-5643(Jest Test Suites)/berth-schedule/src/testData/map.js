const map = {
    "open": false,
    "vessels": {},
    "subscriptions": []
};
export { map };