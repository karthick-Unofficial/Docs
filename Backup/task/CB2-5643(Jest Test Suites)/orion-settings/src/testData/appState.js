const appState = {
    "errors": {
        "newUserErrorMessage": null,
        "passwordChangeError": null,
        "shareDialogError": null,
        "createOrgError": null
    },
    "dialog": {
        "openDialog": null,
        "dialogData": null
    },
    "dock": {
        "dockData": {
            "WavCam": {}
        }
    },
    "sharingTokens": {
        "enabled": true
    },
    "viewing": {
        "selectedEntity": {
            "type": "org",
            "id": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b"
        }
    }
};
export { appState };