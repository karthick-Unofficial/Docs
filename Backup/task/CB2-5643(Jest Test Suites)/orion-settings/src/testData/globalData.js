const globalData = {
    "orgs": {},
    "users": {
        "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700": {
            "admin": true,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T08:00:18.692Z",
            "ecoAdmin": false,
            "email": "testorg@gm.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "c1bc0cd8-9dc8-4dbb-99c1-a0300aa05700",
            "lastModifiedDate": "2023-01-13T08:05:16.333Z",
            "lockedUntil": null,
            "name": "Test Org",
            "orgId": "test_organization_1",
            "orgRole": {
                "roleId": null,
                "title": "Org Admin"
            },
            "passwordChangedDate": "2023-01-13T08:02:06.726Z",
            "passwordExpires": "2023-04-13T08:02:06.726Z",
            "profileImage": "03ba4d40-9319-11ed-b28d-4bd7f7aa52bf",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "test_organization_1_org_user",
            "setPasswordToken": null,
            "username": "testorg@gm.com"
        },
        "d7d232cb-a192-4290-83b8-a70e744a27b0": {
            "admin": false,
            "appSettings": {
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                }
            },
            "authProviderId": "system",
            "contact": {
                "cellPhone": null,
                "officePhone": null
            },
            "createdDate": "2023-01-13T05:01:08.255Z",
            "ecoAdmin": false,
            "email": "s.karthick@rifluxyss.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "d7d232cb-a192-4290-83b8-a70e744a27b0",
            "lastModifiedDate": "2023-01-23T10:57:40.656Z",
            "lockedUntil": null,
            "name": "Karthick S",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "orgRole": {
                "roleId": "org-user",
                "title": "Org User"
            },
            "passwordChangedDate": "2023-01-13T05:10:07.858Z",
            "passwordExpires": "2023-04-13T05:10:07.858Z",
            "profileImage": "c192ce50-9b0c-11ed-beb9-2dd4a7b49e21",
            "resetExpires": null,
            "resetToken": null,
            "role": "Org User",
            "roleId": "ares_security_corporation_org_user",
            "setPasswordToken": null,
            "username": "s.karthick@rifluxyss.com"
        },
        "9610acee-dd03-479c-ab02-5f8fcdb0409f": {
            "admin": true,
            "appSettings": {
                "locale": null,
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "email": "orgadmin@aressecuritycorp.com",
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": false,
            "email": "orgadmin@aressecuritycorp.com",
            "id": "9610acee-dd03-479c-ab02-5f8fcdb0409f",
            "name": "Org Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "role": "org-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "orgadmin@aressecuritycorp.com"
        },
        "ed3d2e80-9737-40cd-ad06-74dc8c6e2003": {
            "admin": true,
            "appSettings": {
                "locale": "en",
                "spotlightProximity": {
                    "unit": "mi",
                    "value": 0.25
                },
                "trackHistory": {
                    "duration": 30
                },
                "tts": {
                    "enabled": false,
                    "type": null
                },
                "unitsOfMeasurement": {
                    "coordinateSystem": "decimal-degrees",
                    "landUnitSystem": "imperial"
                }
            },
            "authProviderId": "system",
            "contact": {
                "address": "1501 Main Street, Suite 120 Columbia, SC 29201",
                "cellPhone": null,
                "email": "ecoadmin@aressecuritycorp.com",
                "officePhone": null,
                "phone": "803-339-0697"
            },
            "disabled": false,
            "ecoAdmin": true,
            "email": "ecoadmin@aressecuritycorp.com",
            "failedLoginAttempts": 0,
            "firstUseAck": true,
            "id": "ed3d2e80-9737-40cd-ad06-74dc8c6e2003",
            "lastModifiedDate": "2023-01-23T06:53:35.286Z",
            "lockedUntil": null,
            "name": "Support Admin",
            "orgId": "ares_security_corporation_f2de9e6b-7f8b-4de6-b1a7-b902e083e11b",
            "profileImage": "a83a6f20-9aea-11ed-9228-07434b76e195",
            "role": "eco-admin",
            "roleId": "ares_security_corporation_org_admin",
            "username": "ecoadmin@aressecuritycorp.com"
        }
    }
};
export { globalData };