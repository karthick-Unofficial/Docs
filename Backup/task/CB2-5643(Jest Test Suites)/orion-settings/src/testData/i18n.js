const i18n = {
	"locale": "en",
	"translations": {}
};
export { i18n };