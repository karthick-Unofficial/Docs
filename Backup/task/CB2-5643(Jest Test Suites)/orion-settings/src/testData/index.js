export { appState } from "./appState";
export { session } from "./session";
export { i18n } from "./i18n";
export { globalData } from "./globalData";