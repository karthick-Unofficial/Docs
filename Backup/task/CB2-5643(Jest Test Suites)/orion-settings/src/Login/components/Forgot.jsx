import React, { useEffect, useState } from "react";
import { useNavigate, useOutletContext } from "react-router-dom";

import { Button, Input } from "@mui/material";

import { Link } from "react-router-dom";
import { Translate, getTranslation } from "orion-components/i18n";
import { useSelector } from "react-redux";

const Forgot = () => {

	const { errorMessage, onForgotClick, resetError } = useOutletContext();
	const navigate = useNavigate();

	//translations here initiates component rerender on successful translations updates.
	const translations = useSelector(state => state.i18n.translations);

	const [usernameValue, setUsernameValue] = useState("");
	const [errMessage, setErrMessage] = useState(errorMessage || "");

	useEffect(() => {
		return () => { resetError(); };
	}, []);

	useEffect(() => {
		setErrMessage(errorMessage);
	}, [errorMessage]);

	const handleUsernameChange = e => {
		setUsernameValue(e.target.value);
	};

	const handleClick = event => {
		if (usernameValue === "") {
			setErrMessage(getTranslation("login.forgotErrText"));
			return;
		}

		setErrMessage("");

		onForgotClick(usernameValue, navigate);
	};

	const handleKeyDownSubmit = e => {
		if (e.keyCode === 13) {
			handleClick();
		}
	};



	const inputStyles = {
		width: "calc(100% - 4rem)",
		paddingTop: "10px",
		paddingBottom: "15px",
		fontSize: 16,
		color: "white"
	};

	const buttonStyles = {
		height: "50px",
		color: "#fff",
		fontSize: "16px",
		fontWeight: "bold",
		lineHeight: "50px"
	};


	return (
		<div>
			<div className="fields-outer-wrapper">
				<span className="text">
					<Translate value="login.recoveryText" />
				</span>
				<div className="forgot fields-wrapper">
					<Input
						id="forgotField"
						type="text"
						placeholder={getTranslation("login.emailAddress")}
						autoFocus={true}
						disableUnderline={true}
						onChange={handleUsernameChange}
						onKeyDown={handleKeyDownSubmit}
						value={usernameValue}
						style={inputStyles}
						className="text-field"

					/>
				</div>
				<Link to="/login" className="forgot-password-link">
					<Translate value="login.backToLogin" />
				</Link>
			</div>
			{errMessage && (
				<p className="error-message">{errMessage}</p>
			)}
			<Button
				fullWidth={true}
				style={buttonStyles}
				onClick={event => handleClick(event)}
			>

				<Translate value="login.sendButton" />

			</Button>

		</div >
	);

};

export default Forgot;