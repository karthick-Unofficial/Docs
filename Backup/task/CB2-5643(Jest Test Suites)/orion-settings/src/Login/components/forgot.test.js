import React from "react";
import Forgot from "./Forgot";
import { shallow } from "enzyme";
import * as reactRedux from 'react-redux';
import { i18n } from "../../testData";

jest.mock('react-router-dom', () => ({
	useOutletContext: () => ({
		errorMessage: ""
	}),
	useNavigate: jest.fn()
}));

jest.mock('orion-components', () => ({
	getTranslation: () => { }
}));

jest.mock("react-redux", () => ({
	useSelector: jest.fn(),
	useDispatch: jest.fn(),
}));

describe("Forgot", () => {
	beforeEach(() => {
		useDispatchMock.mockImplementation(() => () => { });
		useSelectorMock.mockImplementation(selector => selector(mockStore));
	})
	afterEach(() => {
		useDispatchMock.mockClear();
		useSelectorMock.mockClear();
	})

	const useSelectorMock = reactRedux.useSelector;
	const useDispatchMock = reactRedux.useDispatch;

	const mockStore = {
		i18n
	};

	it("renders", () => {
		const wrapper = shallow(<Forgot />);
		expect(wrapper).toMatchSnapshot();
	});
});