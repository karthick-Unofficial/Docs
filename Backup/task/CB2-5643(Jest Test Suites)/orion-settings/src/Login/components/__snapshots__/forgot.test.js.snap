// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forgot renders 1`] = `ShallowWrapper {}`;
